### PR TITLE
Migrate Forno instances to Kubernetes. 

### PR DESCRIPTION
**Pull request by jcortejoso**
### Description

This PR includes different changes needed for the migration from geth full-nodes VM based to K8s Pods. In particular:
- Archive fullnodes moved from VMs to K8s in mainnet.
- Added a new context `GCP_PRIVATE_TXNODES` for the archive full-nodes cluster in mainnet. This context is deployed to the top-level cluster with release name `rc1-fullnodes` (a forward PR could include the capability of customizing the celo-fullnode deployment name, allowing the deployment of multiple fullnode chart releases in the same cluster).
- Added the contexted variables :

1. `<CTX>_FULL_NODES_RPC_API_METHODS`: Customize the rpc and ws allowed methods in the celo-fullnodes.
2. `<CTX>_FULL_NODES_GETH_GC_MODE`: Customize geth gc mode.
3. `<CTX>_FULL_NODES_USE_GSTORAGE_DATA`: Enable the chain restoring option or not. Currently there is not available backup for archive nodes.

- Modified blockscout setup to allow custom rpc/ws endpoints using the variables `BLOCKSCOUT_OVERRIDE_RPC_ENDPOINT` and `BLOCKSCOUT_OVERRIDE_WS_ENDPOINT`. This is needed because archive nodes setup in mainnet is not based on `testnet` helm package but `celo-fullnode` helm package.
- Updated forno terraform module to terraform>=1.13.
- Added `rc1-forno.celo-testnet.org.` as domain to (new)forno domains.
-  Added `FORNO_BANNED_CIDR` to ban CIDRs to use forno.

### Other changes

- Adding a k8s service to prometheus setup.
- Updated some docker images to node:12.

### Tested

Most of the changes are already deployed.

### Related issues

- Fixes celo-blockchain#1393

### Backwards compatibility

Only compatibility problem I can think about is if someone is using old forno ip directly. 

### Pending work

- [x] Update the env files to add the new contexted variables to the existing contexts
- [x] Apply the terraform changes. 
- [ ] Update blockscout deployment to start using the k8s archive nodes

### Documentation

_The set of community facing docs that have been added/modified because of this change_

### CODE REVIEW

File: .env.rc1 [hunk 1]
```diff
@@ -191,6 +214,22 @@ GCP_FORNO_SOUTHAMERICA_EAST1_FULL_NODES_DISK_SIZE=100
...
 # suffix of the static nodes file, e.g. `<env>.<suffix>`
 GCP_FORNO_SOUTHAMERICA_EAST1_FULL_NODES_STATIC_NODES_FILE_SUFFIX=gcp-southamerica-east1
+GCP_FORNO_SOUTHAMERICA_EAST1_FULL_NODES_RPC_API_METHODS="eth,net,rpc,web3"
+GCP_FORNO_SOUTHAMERICA_EAST1_FULL_NODES_GETH_GC_MODE="full"
+GCP_FORNO_SOUTHAMERICA_EAST1_FULL_NODES_USE_GSTORAGE_DATA=true
+
+
+# ---- Private txnodes context ----
+
+GCP_PRIVATE_TXNODES_GCP_PROJECT_NAME=celo-testnet-production
```
Comment (trianglesphere) [inline .env.rc1:224]: Do we want to move all testnets to k8s archive nodes?

File: packages/celotool/src/lib/blockscout.ts [hunk 2]
```diff
@@ -129,16 +129,26 @@ async function helmParameters(
...
     const txNodeLbIp = await getInternalTxNodeLoadBalancerIP(celoEnv)
     params.push(`--set blockscout.jsonrpc_http_url=http://${txNodeLbIp}:8545`)
     params.push(`--set blockscout.jsonrpc_ws_url=ws://${txNodeLbIp}:8546`)
   } else if (privateNodes > 0) {
     params.push(`--set blockscout.jsonrpc_http_url=http://tx-nodes-private:8545`)
     params.push(`--set blockscout.jsonrpc_ws_url=ws://tx-nodes-private:8546`)
   } else {
-    params.push(`--set blockscout.jsonrpc_http_url=http://tx-nodes-headless:8545`)
-    params.push(`--set blockscout.jsonrpc_ws_url=ws://tx-nodes-headless:8546`)
+    params.push(`--set blockscout.jsonrpc_http_url=http://tx-nodes:8545`)
```
Comment (trianglesphere) [inline packages/celotool/src/lib/blockscout.ts:150]: I'm not seeing where we went from `tx-nodes-headless` to `tx-nodes`. (Though I'm also not familiar with why/how this lookup works).

File: packages/celotool/src/lib/k8s-fullnode/base.ts [hunk 3]
```diff
@@ -94,21 +97,26 @@ export abstract class BaseFullNodeDeployer {
       )
     }
 
-    const rpcApis = 'eth,net,rpc,web3'
+    // const rpcApis = 'eth,net,rpc,web3'
```
Comment (trianglesphere) [inline packages/celotool/src/lib/k8s-fullnode/base.ts]:
```suggestion
```

**Review by trianglesphere — COMMENTED** [review]
One small suggestion & a couple questions, but this is looking good.

File: .env.rc1 [hunk 4]
```diff
@@ -27,6 +27,8 @@ BLOCKSCOUT_SUBNETWORK_NAME="Celo"
 BLOCKSCOUT_METADATA_CRAWLER_IMAGE_REPOSITORY="gcr.io/celo-testnet/celo-monorepo"
 BLOCKSCOUT_METADATA_CRAWLER_IMAGE_TAG="metadata-crawler-007f29dc8ebf837543e6e28a14818076e88c69cd"
 BLOCKSCOUT_METADATA_CRAWLER_SCHEDULE="0 */2 * * *"
+BLOCKSCOUT_OVERRIDE_RPC_ENDPOINT="http://rc1-fullnodes-rpc.rc1.svc.cluster.local:8545"
```
Comment (carterqw2) [inline .env.rc1:30]: It looks like session affinity is not configured for this service, was that intentional?

Comment (jcortejoso) [reply .env.rc1:30]: True, thanks for noticing.

File: .env.rc1 (hunk 1 above)
Comment (jcortejoso) [reply .env.rc1:224]: I'd say we do not want, at least I cannot think any reason to migrate current archive nodes included with `testnet` package to `archive` nodes based on the `celo-fullnode` package, in any other environment except to mainnet.
Mainnet cluster/deployment is special as there are not validators or regular tx-nodes, so to deploy only fullnodes using `testnet` required some changes on current code and seemed more appropriate using the celo-fullnode package. But also I think both solutions seems in the end a bit artificial (also related right now we have "contexted" commands/packages and non-contexted commands/packages).

File: packages/celotool/src/lib/blockscout.ts (hunk 2 above)
Comment (jcortejoso) [reply packages/celotool/src/lib/blockscout.ts:150]: For headless service, `kube-dns` returns a list of the pod's ips that are endpoints for that service, and the service does not have a service ip. In the regular service, the service has a cluster ip, kube-dns returns that ip for resolving the service, and `kube-proxy` routes the cluster-ip to a pod endpoint using iptables.
We have been switching between `headless` and regular services trying to improve the session affinity, and I think we have conclude is better to use regular services configured with sessionAffinity.

**Review by carterqw2 — APPROVED** [review]

**Pull request merged**